### DISCUSSION TITLE
Clarification on accumulated values

### DISCUSSION THREAD

**Discussion by guidocioni** [open-meteo/open-meteo#552] [Q&A]
We know that all model data is interpolated to 1 hourly frequency to ease comparison between models. However, the original data may have a less fine granularity, e.g. GFS Ensemble only have data every 3 hours. 

For variables that have an instantaneous values, like 2m Temperature, there is no issue.
However, I'm not sure I understand how this was done for variables that have an accumulation period. For example, precipitation coming from GFS Ensemble is accumulated over the preceding 3 hrs, because this is the output period of the model. When interpolating to 1 hourly data did you de-accumulate by just splitting the 3-hourly value into 3 equal values? 

I'm asking because, for visualisation purposes, I'm trying to decimate again the data to the "original" time resolution, so I'm wondering what would be the easiest and most correct way of doing this.
Right now I'm doing something on the line of

```python
data.resample('3H', on='time').first().reset_index()
```

but this obviously will only work for istantaneous values.
I think that I need to do some explicit sum (instead than just using `first`) to retrieve the "original" values, but I'm wondering if there's a better way since the ensemble variables are many...

**Comment by patrick-zippenfenig** [comment]
Precipitation sum is divided by 3 in case of interpolation from 3 to 1-hourly data. For each weather variables, different interpolation techniques are selected. The most complicated is solar radiation to which needs to be adjusted to the solar position.

I am considering to add an URL parameter to select 3 or even 6 hourly data. Unfortunately, it will be a bit more work because there is no code to aggregate from 1h to 3 yet. Feel free to open a issue ticket with a feature request

**Reply by guidocioni** [reply]
Ok, thanks for the clarification. So I guess the same logic is used for rain or snowfall right?

**Reply by patrick-zippenfenig** [reply]
Yes. Each variable has different interpolation methods assigned: https://github.com/open-meteo/open-meteo/blob/2d87f2cde317f4b82dd7fbe7b62f266e18e34809/Sources/App/Gfs/GfsVariable.swift#L142-L229
